R/match_loanbooks.R

```diff
@@ -143,21 +143,16 @@ match_loanbooks <- function(config) {
 
     ## match data----
     if (matching_use_manual_sector_classification) {
-      withr::with_options(
-        new = list(r2dii.match.sector_classifications = sector_classification_system),
-        code = {
-          getOption("r2dii.match.sector_classifications")
-          matched_lbk_i <- r2dii.match::match_name(
-            loanbook = raw_lbk[[i]],
-            abcd = abcd,
-            by_sector = matching_by_sector,
-            min_score = matching_min_score,
-            method = matching_method,
-            p = matching_p,
-            overwrite = matching_overwrite,
-            join_id = matching_join_id
-          )
-        }
+      matched_lbk_i <- r2dii.match::match_name(
```

Comment:
nit: instead of this big conditional block here, we could now easily just always pass the argument `sector_classification = sector_classification_system`, if we introduce else logic further above where we load the sector classification system. if no manual file is provided we assign the r2dii.data version.

Reply:
totally agree.... I went for easiest to review/minimal change, but also saw that the conditional is not really needed anymore.... I can update this PR, or you can if you want to / get to it

Reply:
done. I might update the cookbook while at it

Reply:
oh, interesting.... yeah, we're now in that loop of relying on a release version of another package

Reply:
yes, unfortunately. which means we need to bump the required version of `r2dii.match` in the `DESCRIPTION` soon too

Reply:
could temporarily set `Remotes` for it to get the dev version

but one thing I definitely forgot to do here is pin the `r2dii.match` version to >=0.3.0 (?) in DESCRIPTION

Reply:
realistically, I think the most appropriate approach would be to wait until `r2dii.match` v0.3.0 is released on CRAN, then make sure this PR includes the >=v0.3.0  requirement in DESCRIPTION

Reply:
Relates to https://github.com/RMI-PACTA/r2dii.match/issues/510